### PR TITLE
Make room_fixture less magical.

### PR DESCRIPTION
**Pull request by NegativeMjark**
Rather than calling ``matrix_create_and_join_room`` which does extra stuff it
calls ``matrix_create_room``. There are a couple of tests that depend on
the extra stuff so I've added a ``magic_room_fixture`` which calls
``matrix_create_and_join_room`` preserving the old behaviour.

### CODE REVIEW

File: tests/50federation/30room-join.pl [hunk 1]
```diff
@@ -116,11 +116,11 @@
 test "Inbound federation can receive room-join requests",
    requires => [ $main::OUTBOUND_CLIENT, $main::INBOUND_SERVER,
                  $main::HOMESERVER_INFO[0],
-                 room_fixture( requires_users => [ local_user_fixture() ] ),
+                 local_user_and_room_fixtures(),
                  federation_user_id_fixture() ],
 
    do => sub {
-      my ( $outbound_client, $inbound_server, $info, $room_id, $user_id ) = @_;
+      my ( $outbound_client, $inbound_server, $info, $creator, $room_id, $user_id ) = @_;
```
Comment (leonerd) [inline tests/50federation/30room-join.pl]: Since `$creator` isn't actually used by the test, you can just assign it into `undef` instead to leave a hole

File: tests/50federation/32room-getevent.pl [hunk 2]
```diff
@@ -1,10 +1,10 @@
 test "Inbound federation can return events",
    requires => [ $main::OUTBOUND_CLIENT, $main::HOMESERVER_INFO[0],
-                 room_fixture( requires_users => [ local_user_fixture() ] ),
+                 local_user_and_room_fixtures(),
                  federation_user_id_fixture() ],
 
    do => sub {
-      my ( $outbound_client, $info, $room_id, $user_id ) = @_;
+      my ( $outbound_client, $info, $creator, $room_id, $user_id ) = @_;
```
Comment (leonerd) [inline tests/50federation/32room-getevent.pl]: Ditto

File: tests/50federation/33room-get-missing-events.pl [hunk 3]
```diff
@@ -101,11 +101,11 @@
 
 test "Inbound federation can return missing events",
    requires => [ $main::OUTBOUND_CLIENT, $main::HOMESERVER_INFO[0],
-                 room_fixture( requires_users => [ local_user_fixture() ] ),
+                 local_user_and_room_fixtures(),
                  federation_user_id_fixture() ],
 
    do => sub {
-      my ( $outbound_client, $info, $room_id, $user_id ) = @_;
+      my ( $outbound_client, $info, $creator, $room_id, $user_id ) = @_;
```
Comment (leonerd) [inline tests/50federation/33room-get-missing-events.pl]: Ditto

File: tests/10apidoc/33room-members.pl [hunk 4]
```diff
@@ -408,15 +408,38 @@ sub matrix_create_and_join_room
...
+   );
+}
+
+push @EXPORT, qw( magic_room_fixture );
+
+sub magic_room_fixture
+{
    my %args = @_;
 
+   my $magic = delete $args{magic};
```
Comment (leonerd) [inline tests/10apidoc/33room-members.pl]: I've looked this up and down and I don't see any use of `$magic` here, nor any calls passing it in. Could this line be deleted therefore?

File: tests/10apidoc/33room-members.pl [hunk 5]
```diff
@@ -408,15 +408,38 @@ sub matrix_create_and_join_room
...
+   my ( $user_fixture, %args ) = @_;
+
+   fixture(
+      requires => [ $user_fixture ],
+
+      setup => sub {
+         my ( $user ) = @_;
+
+         matrix_create_room( $user, %args )->then( sub {
+            my ( $room_id ) = @_;
```
Comment (leonerd) [inline tests/10apidoc/33room-members.pl]: I was initially unsure why this `then` block exists, but I see it's because of the 2-value return from `matrix_create_room` that also passes the alias back. Could you pop a little comment in here just reminding people of such, so nobody's tempted to remove the block and thus upset most of the tests?

File: tests/50federation/30room-join.pl (hunk 1 above)
Comment (NegativeMjark) [inline tests/50federation/30room-join.pl]: Done d43e3b9

File: tests/50federation/32room-getevent.pl (hunk 2 above)
Comment (NegativeMjark) [inline tests/50federation/32room-getevent.pl]: Done d43e3b9

File: tests/50federation/33room-get-missing-events.pl (hunk 3 above)
Comment (NegativeMjark) [inline tests/50federation/33room-get-missing-events.pl]: Done d43e3b9

File: tests/10apidoc/33room-members.pl (hunk 4 above)
Comment (NegativeMjark) [inline tests/10apidoc/33room-members.pl]: Done d43e3b9

File: tests/10apidoc/33room-members.pl (hunk 5 above)
Comment (NegativeMjark) [inline tests/10apidoc/33room-members.pl]: Done d43e3b9

File: lib/SyTest/Synapse.pm [hunk 6]
```diff
@@ -142,7 +142,7 @@ sub start
          bind_address => "127.0.0.1",
          tls => 1,
          resources => [{
-            names => [ "client", "federation" ], compress => 0
+            names => [ "client", "federation", "replication" ], compress => 0
```
Comment (leonerd) [inline lib/SyTest/Synapse.pm]: Did you mean to put this in? It looks unrelated...

Comment (NegativeMjark) [inline lib/SyTest/Synapse.pm]: Fixed fa64d81

**Comment by leonerd** [comment]
LGTM. Jenkins looks a bit unhappy though.
